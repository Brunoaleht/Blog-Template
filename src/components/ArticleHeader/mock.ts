import { ArticleHeaderProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  title: 'ReactHook Studies',
  slug: 'react-hook-studies',
  cover: {
    coverData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677769491/thumbnail_MKS_entrevista_87872be63c_60bb358adf.jpg',
      },
    },
  },
  excerpt:
    'I made this project using various react hooks, such as useEffect to make a call to the api, use State to keep my data in state, useContext and useReducer to do the work in the general context in the shopping cart.',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleHeaderProps;
