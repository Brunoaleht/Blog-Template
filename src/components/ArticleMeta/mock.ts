import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleMetaProps;
