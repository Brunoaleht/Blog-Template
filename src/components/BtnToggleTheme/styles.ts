import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
  `}
`;
export const Label = styled.label`
  ${() => css`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    overflow: hidden;
    font-size: 0;
    color: rgba(0, 0, 0, 0);
  `}
`;
export const BtnInput = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
    }
    &:focus + ${Slider} {
      border: solid 1px ${theme.colors.secondary};
    }
    &:checked + ${Slider}::before {
      transform: translateX(1.95rem);
      background: ${theme.colors.primary};
    }
  `}
`;
export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    cursor: pointer;
    background: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;
    border: solid 1px ${theme.colors.darkerGray};
    border-radius: 2rem;
    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.textDark};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
