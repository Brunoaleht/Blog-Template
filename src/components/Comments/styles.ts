import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
