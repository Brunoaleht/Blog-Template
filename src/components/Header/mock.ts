import { HeaderProps } from '.';
export default {
  blogName: 'Bruno Alexandre',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus doloremque nulla eius aperiam modi error facere, commodi.',
  logo: {
    logoData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677869914/logo1_6dca30fa4b.png',
      },
    },
  },
  showText: true,
} as HeaderProps;
