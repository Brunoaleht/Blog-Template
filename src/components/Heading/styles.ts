import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
  `,
};

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

export const Title = styled.h1<Pick<HeadingProps, 'colorDark' | 'size' | 'uppercase'>>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primary : theme.colors.white};
    line-height: 1.2;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
