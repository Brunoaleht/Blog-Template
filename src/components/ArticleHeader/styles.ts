import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';
export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.large};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.fonts.sizes.medium};
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
