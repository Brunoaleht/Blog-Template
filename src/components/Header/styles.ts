import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.textDark};
    font-size: ${theme.fonts.sizes.small};
    margin: 0 auto;
    ${HeadingStyle} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }
    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeadingStyle} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
    p {
      margin-top: 0;
    }
    @media ${theme.media.lteSmall} {
      margin-left: 0;
    }
  `}
`;
