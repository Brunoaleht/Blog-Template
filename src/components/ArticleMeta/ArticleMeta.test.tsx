import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from '.';
import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category link', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByRole('link', { name: 'Bruno Alexandre' })).toHaveAttribute('href', '/author/bruno-alexandre');
    expect(screen.getByRole('link', { name: 'ReactJs' })).toHaveAttribute('href', '/category/react-js');
    expect(screen.getByRole('link', { name: 'JavaScript' })).toHaveAttribute('href', '/category/java-script');
  });
  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);
    expect(screen.getByText('2 de mar. de 2021')).toHaveAttribute('dateTime', props.createdAt);
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);
    expect(container).toMatchSnapshot();
  });
});
