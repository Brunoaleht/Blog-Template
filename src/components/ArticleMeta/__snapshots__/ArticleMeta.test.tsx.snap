// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por: 
      </span>
      <a
        href="/author/bruno-alexandre"
      >
        Bruno Alexandre
      </a>
      <span
        class="separador"
      >
         | 
      </span>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
      <span
        class="separador"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react-js"
          >
            ReactJs
          </a>
        </span>
        <span>
          <a
            href="/category/java-script"
          >
            JavaScript
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;
